WooCommerce/Classes/POS/TabBar/POSTabCoordinator.swift

```diff
@@ -143,7 +142,36 @@ private extension POSTabCoordinator {
     func presentPOSView(siteID: Int64) {
         Task { @MainActor [weak self] in
             guard let self else { return }
-            let serviceAdaptor = POSServiceLocatorAdaptor()
+
+            // Check local catalog eligibility before initializing infrastructure
+            // Try to use pre-created service from eligibility checker, otherwise create it now
+            let eligibilityService: POSLocalCatalogEligibilityServiceProtocol
+            if let preCreatedService = (eligibilityChecker as? POSTabEligibilityChecker)?.localCatalogEligibilityService {
```

Comment:
We could include it in `POSEntryPointEligibilityCheckerProtocol`, or even use `POSTabEligibilityChecker` directly in `POSTabCoordinator`. This coordinator doesn't rely on protocols. It's like a factory that builds and uses concrete types.